graphql/schemabuilder/reflect.go

```diff
@@ -507,6 +518,9 @@ func (sb *schemaBuilder) buildFunction(typ reflect.Type, fun reflect.Value) (*gr
 			var result interface{}
 			if hasRet {
 				result = out[0].Interface()
+				if out[0].Kind() == reflect.Ptr && out[0].IsNil() && !m.MarkedNullable {
```

Comment:
I was previously in the mindset that "structs are safe" to pass around given their lack of `nil` (akin to null checking in ts/flow), but hadn't considered that dereferencing a pointer type (would be common with sqlgen code) is still not a statically checked operation in golang.